### DISCUSSION TITLE
Can you help me with a script?

### DISCUSSION THREAD

**Discussion by gerroon** [TriliumNext/Trilium#6335] [Q&A]
Hi

I know nothing about js at all so I am wondering if someone is willing to modify the existing date script so it does what I am looking for?

I would like another `today` button/script that creates a note named `DD-MM-YYYY` under the currently selected note? 

The current `today` script works good for journaling fashion but I am looking for that modified version.

**Comment by gerroon** [comment]
Is this possible at all?

**Comment by zadam** [comment]
> Is this possible at all?

Yes. You can use `api.getActiveTabNote()` in frontend API and then use that as a parent in backend's `api.createNewNote()`.

**Comment by gerroon** [comment]
Thanks but I know rabiit poop about js :( It would be nice if someone can give me the full script with the necessary css I guess.

thanks

**Comment by wouterbatelaan-es** [comment]
This seems to work for me:
```
api.addButtonToToolbar({
    title: 'Subday',
    icon: 'calendar',
    action: async function() {
        const activeNote = await api.getActiveTabNote();
        const noteId = await api.runOnServer(
            async (parentId) => {
                const title = api.dayjs().format("YYYY-MM-DD");
                const {note} = await api.createNote(parentId, title);
                return note.noteId;
            },
        [activeNote.noteId]);

        await api.waitUntilSynced();
        
        api.activateNote(noteId);
    }
});
```
(using different date format - easier on sorting)

**Comment by gerroon** [comment]
Hey, thanks for your help. This is exactly what I was looking for.

> This seems to work for me:
> 
> ```
>     }
> });

**Comment by gerroon** [comment]
@wouterbatelaan-es 

Hey this script works great except that it does not seem to be executed automatically every time I restart Trilium. D o you know how I can make it run automatically everytime I run Trilium?

**Comment by wouterbatelaan-es** [comment]
Hi, you need to add an attribute to it.
`#run=frontendStartup`
For an example see the Today note.
